cpp/include/ray/api/object_ref.h

```diff
@@ -46,6 +47,17 @@ ObjectRef<T>::ObjectRef() {}
 template <typename T>
 ObjectRef<T>::ObjectRef(const ObjectID &id) {
   id_ = id;
+  if (CoreWorkerProcess::IsInitialized()) {
+    auto &core_worker = CoreWorkerProcess::GetCoreWorker();
+    core_worker.AddLocalReference(id_);
+  }
+}
+template <typename T>
+ObjectRef<T>::~ObjectRef() {
+  if (CoreWorkerProcess::IsInitialized()) {
+    auto &core_worker = CoreWorkerProcess::GetCoreWorker();
+    core_worker.RemoveLocalReference(id_);
```

Comment:
Won't this be called at the end of the function and the reference will be removed? Shouldn't we remove those when the driver detaches?

Reply:
Maybe this is true! I tried to emulate the `ObjectRef` in python. Python `ObjectRef` removes local reference [here](https://github.com/ray-project/ray/blob/master/python/ray/includes/object_ref.pxi#L22-L26).